packages/components/src/presentation-components/instance-filter-builder/PresentationInstanceFilterDialog.tsx

```diff
@@ -66,81 +68,95 @@ export function PresentationInstanceFilterDialog(props: PresentationInstanceFilt
+      disposed.current = true;
     };
-  }, [descriptorGetter]);
+  }, [descr, descriptor]);
 
   return descriptor;
 }
 
-function DelayedCenteredProgressRadial() {
-  const [show, setShow] = useState(false);
+interface PresedntationInstanceFilterDialogContentProps extends Omit<PresentationInstanceFilterDialogProps, "isOpen" | "title" | "descriptor"> {
+  descriptor: Descriptor;
+}
 
-  useEffect(() => {
-    const timeout = setTimeout(() => {
-      setShow(true);
-    }, 250);
-    return () => {
-      clearTimeout(timeout);
-    };
-  }, []);
+function PresentationInstanceFilterDialogContent(props: PresedntationInstanceFilterDialogContentProps) {
+  const { onApply, initialFilter, descriptor, imodel, ruleGroupDepthLimit, filterResultCountRenderer, onClose } = props;
+
+  const { rootGroup, actions, validate } = usePropertyFilterBuilder({
+    initialFilter: initialFilter ? convertPresentationFilterToPropertyFilter(descriptor, initialFilter.filter) : undefined,
+  });
+
+  const filteringProps = usePresentationInstanceFilteringProps(descriptor, imodel, initialFilter?.usedClasses);
+
+  const navigationPropertyEditorContextValue = useFilterBuilderNavigationPropertyEditorContext(imodel, descriptor);
+
+  const applyButtonHandle = () => {
+    const filter = validate();
+    if (filter) {
+      const presentationInstanceFilter = createPresentationInstanceFilter(descriptor, filter);
+      if (presentationInstanceFilter) {
+        onApply({ filter: presentationInstanceFilter, usedClasses: filteringProps.selectedClasses });
+      }
+    }
+  };
+
+  const isDisabled = !rootGroup.items
+    .flat(ruleGroupDepthLimit ?? Number.MAX_SAFE_INTEGER)
+    .some((item) => !isPropertyFilterBuilderRuleGroup(item) && item.operator !== undefined);
 
-  if (!show) {
-    return null;
-  }
   return (
+    <>
+      <Dialog.Content className="presentation-instance-filter-content">
+        <navigationPropertyEditorContext.Provider value={navigationPropertyEditorContextValue}>
+          <InstanceFilterBuilder {...filteringProps} rootGroup={rootGroup} actions={actions} />
+        </navigationPropertyEditorContext.Provider>
+      </Dialog.Content>
+      <div className="presentation-instance-filter-dialog-bottom-container">
+        <div>{filterResultCountRenderer && filterResultCountRenderer()}</div>
+        <Dialog.ButtonBar className="presentation-instance-filter-button-bar">
+          <Button className="presentation-instance-filter-dialog-apply-button" styleType="high-visibility" onClick={applyButtonHandle} disabled={isDisabled}>
+            {translate("instance-filter-builder.apply")}
+          </Button>
+          <Button className="presentation-instance-filter-dialog-close-button" onClick={onClose}>
+            {translate("instance-filter-builder.cancel")}
+          </Button>
+        </Dialog.ButtonBar>
+      </div>
+    </>
+  );
+}
+
+function DelayedCenteredProgressRadial() {
+  const show = useDelay();
+
+  return show ? (
```

Comment:
Why was this changed to ternary operator?

Reply:
reverted